### DISCUSSION TITLE
Discord Native support design discussion.

### DISCUSSION THREAD

**Discussion by J-Tanzanite** [J-Tanzanite/Little-Anti-Cheat#182] [General]
As discussed in a previous thread (https://github.com/J-Tanzanite/Little-Anti-Cheat/issues/154), it would be nice to have Discord support.
But given that there are multiple discord APIs to choose from, it's better to have Discord support come in the form of a third party plugin.

There are forwards in place in Lilac that are called when a cheater is detected, but it carries no detection info, which might be useful to have in a Discord message.
And it would also come in handy in this plugin, with an option to avoid recording autoshoot cases: https://github.com/J-Tanzanite/Lilac-SourceTV

The easiest and best way to go about this is to add a native function to Lilac that other plugins can call to get more information about a detection; rather than sending an entire log line worth of information.

A possible design for this would be to just do it this way (Numbers not set in stone for anything):
```sp
#define LILAC_INFO_AIMBOT_DETECTION    1000 /* How many detections does the client have. */
#define LILAC_INFO_AIMBOT_DELTA        1001 /* Largest angle delta. */
#define LILAC_INFO_AIMBOT_TOTAL_DELTA  1002 /* Amount of angle change in 0.5 seconds. */
#define LILAC_INFO_AIMBOT_FLAGS        1003 /* Aimbot detection flags (AimSnap, Autoshoot, etc). */

#define LILAC_INFO_AIMLOCK_DETECTION  2000 /* How many detections does the client have. */

#define LILAC_INFO_ANGLE_PITCH  3000 /* Pitch angle. */
#define LILAC_INFO_ANGLE_ROLL   3001 /* Roll angle. */
#define LILAC_INFO_ANGLE_YAW    3002 /* Yaw angle. */

#define LILAC_INFO_BHOP_DETECTION  5000 /* How many detections does the client have. */
#define LILAC_INFO_BHOP_BHOPS      5001 /* How many perfect consecutive BHops were done. */
#define LILAC_INFO_BHOP_JUMPTICKS  5002 /* How many jump presses. */

#define LILAC_INFO_MACRO_DETECTION  8000 /* How many detections does the client have. */
#define LILAC_INFO_MACRO_TYPE       8001 /* Type of macro detected. */
#define LILAC_INFO_MACRO_AMOUNT     8002 /* How many key presses were detected. */

/* Returns true if a 'lilac_info' number is valid, false if depreciated, old Lilac version, etc... */
native bool lilac_info_valid(int lilac_info);

/* Returns a value (can be a float, int, etc) based on the requested lilac_info number from a client. */
native any lilac_info_fetch(int client, int lilac_info);
```

Please note that this is just a suggestion for an implementation I may add.
I'd love some feedback.

**Comment by PorkBellyButton** [comment]
Would be amazing
